operator/src/main/java/oracle/kubernetes/operator/helpers/EventHelper.java

```diff
@@ -119,39 +145,90 @@ private boolean isDuplicatedStartedEvent(DomainPresenceInfo info) {
     private boolean hasProcessingNotStarted(DomainPresenceInfo info) {
       return Optional.ofNullable(info).map(dpi -> dpi.getLastEventItem() != DOMAIN_PROCESSING_STARTING).orElse(false);
     }
+
+    private class CreateEventResponseStep extends ResponseStep<V1Event> {
+      CreateEventResponseStep(Step next) {
+        super(next);
+      }
+
+      @Override
+      public NextAction onSuccess(Packet packet, CallResponse<V1Event> callResponse) {
+        Optional.ofNullable(packet.getSpi(DomainPresenceInfo.class))
+            .ifPresent(dpi -> dpi.setLastEventItem(eventData.eventItem));
+        return doNext(packet);
+      }
+    }
+
+    private class ReplaceEventResponseStep extends ResponseStep<V1Event> {
+
+      ReplaceEventResponseStep(Step next) {
+        super(next);
+      }
+
+      @Override
+      public NextAction onSuccess(Packet packet, CallResponse<V1Event> callResponse) {
+        Optional.ofNullable(packet.getSpi(DomainPresenceInfo.class))
+            .ifPresent(dpi -> dpi.setLastEventItem(eventData.eventItem));
+        return doNext(packet);
+      }
+
+      @Override
+      public NextAction onFailure(Packet packet, CallResponse<V1Event> callResponse) {
+        return UnrecoverableErrorBuilder.isAsyncCallNotFoundFailure(callResponse)
+            ? doNext(Step.chain(createCreateEventCall(createEventModel(packet, eventData)), getNext()), packet)
+            : super.onFailure(packet, callResponse);
+      }
+    }
   }
 
-  private static V1Event createEvent(
+  private static V1Event createEventModel(
       Packet packet,
       EventData eventData) {
     DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
-    eventData.namespace(Optional.ofNullable(info)
-        .map(DomainPresenceInfo::getNamespace).orElse(eventData.namespace));
-    eventData.resourceName(eventData.eventItem.calculateResourceName(info, eventData.namespace));
+    EventItem eventItem = eventData.eventItem;
+
+    String namespace = getNamespaceFromInfo(eventData, info);
+    eventData.namespace(namespace);
+    eventData.resourceName(eventItem.calculateResourceName(info, namespace));
+    eventData.domainPresenceInfo(info);
+
     return new V1Event()
         .metadata(createMetadata(eventData))
         .reportingComponent(WEBLOGIC_OPERATOR_COMPONENT)
         .reportingInstance(getOperatorPodName())
-        .lastTimestamp(eventData.eventItem.getLastTimestamp())
-        .type(eventData.eventItem.getType())
-        .reason(eventData.eventItem.getReason())
-        .message(eventData.eventItem.getMessage(eventData.getResourceName(), eventData))
-        .involvedObject(eventData.eventItem.createInvolvedObject(eventData));
+        .lastTimestamp(eventItem.getLastTimestamp())
```

Comment:
Minor: IMO it might be more clear if `getLastTimestamp`  method name is changed to `getCurrentTimestamp`  since lastTimestamp is set to current time.